.github/workflows/ci_tests.yml

```diff
@@ -0,0 +1,60 @@
+name: Unit, Static and other Tests
+
+on:
+  create:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  dataloader_test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: test the dataloader without / with sanitize address
+        run: |
+          cd dev/test
+          make test_dataloader
+          ./test_dataloader   
+          make clean       
+          make test_dataloader TEST_CFLAGS="-fsanitize=address -fno-omit-frame-pointer"
+          ./test_dataloader          
+
+  ptx_and_sass_files:
+    runs-on: ubuntu-latest
+    container:
+      image: nvidia/cuda:12.4.1-devel-ubuntu22.04
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Install OpenMP and OpenMPI
+        run: apt-get update && apt-get install -y libomp-dev libopenmpi-dev
+    
+      - name: Generate ptx/sass files and upload them to persistent storage
+        run: |
+          mkdir -p dev/cuda/ptx_sass_logs
+          make train_gpt2cu
```

Comment:
should we specify which GPUs we want to generate SASS for here?

Reply:
This was @ademeure's request. Any comments?

Reply:
@ngc92 - so for your comment above - had the question below too - would be to add the A100 and H100's SASS in a step here?

Reply:
Added H100 and A100 to ptx and sass file generation